src/bin/pg_autoctl/monitor.c

```diff
@@ -615,9 +614,6 @@ monitor_register_node(Monitor *monitor, char *formation, char *host, int port,
 		return false;
 	}
 
-	/* disconnect from PostgreSQL now */
-	pgsql_finish(&monitor->pgsql);
```

Comment:
We don't call this anywhere anymore now. Shouldn't these calls be placed after `COMMIT`/`ROLLBACK` now?

Reply:
That's a good point, fixed. In the branch where we refactor the main keeper look, I have also added some connection management, and so I guess that's also what I had in mind here, but well, this branch didn't have it. I have added it now, and kept the idea to close the connection after either COMMIT or ROLLBACK of the registration call.